index.js

```diff
@@ -109,6 +169,266 @@ export const immediate = x => Future ((rej, res) => {
+//. object, returns a Future which sends the object to the server at the
+//. given URL after JSON-encoding it.
+//.
+//. ```js
+//. sendJson ('PUT')
+//.          ('https://example.com/users/bob')
+//.          ({Authorization: 'Bearer asd123'})
+//.          ({name: 'Bob', email: 'bob@example.com'})
+//. ```
+export const sendJson = method => url => headers => json => {
+  const buf = Buffer.from (JSON.stringify (json));
+  return send ('application/json') (method) (url) (headers) (buf);
+};
+
+//# sendForm :: Method -> String -> StrMap String -> JsonValue -> Future Error {meta :: MessageMeta, body :: String}
+//.
+//. A version of [`send`](#send) specialized in sending form data.
+//.
+//. Given a request method, a URL, a StrMap of headers and a JavaScript plain
+//. object, returns a Future which sends the object to the server at the
+//. given URL after www-form-encoding it.
+//.
+//. ```js
+//. sendJson ('POST')
+//.          ('https://example.com/users/create')
+//.          ({Authorization: 'Bearer asd123'})
+//.          ({name: 'Bob', email: 'bob@example.com'})
+//. ```
+export const sendForm = method => url => headers => form => {
+  const buf = Buffer.from (qs.stringify (form));
+  return send ('application/x-www-form-urlencoded')
+              (method)
+              (url)
+              (headers)
+              (buf);
+};
+
+//# acceptStatus :: Number -> {meta :: MessageMeta, body :: a} -> Future {meta :: MessageMeta, body :: a} {meta :: MessageMeta, body :: a}
+//.
+//. This function "tags" a buffered response based on a given status code.
+//. If the response status matches the given status code, the returned Future
+//. will resolve. If it doesn't, the returned Future will reject.
+//.
+//. The idea is that you can compose this function with one that returns a
+//. buffered response (such as most functions listed above), and reject any
+//. responses that don't meet the expected status code. In combination with
+//. [`responseToError`](#responsetoerror), you can then flatten it back into
+//. the outer Future.
+//.
+//. The usage example under the [Http](#http)-section shows this.
+export const acceptStatus = code => res => (
+  code === res.meta.code ? resolve (res) : reject (res)
+);
+
+//# responseToError :: {meta :: MessageMeta, body :: String} -> Error
+//.
+//. Given a buffered response, returns an instance of Error with a message
+//. based on the content of the response. See [`acceptStatus`](#acceptstatus)
+//. for instructions on how to use this function.
+//.
+//. ```js
+//. responseToError ({
+//.   meta: {code: 500, message: 'Internal Server Error', headers: {}},
+//.   body: 'Dear user,\n\nEverything broke down.\nWe are sorry.',
+//. });
+//. // ->
+//. new Error (
+//.   'Unexpected Internal Server Error (500) response. Response body:\n' +
+//.   '\n' +
+//.   '  Dear user,\n' +
+//.   '  \n' +
+//.   '  Everything broke down.\n' +
+//.   '  We are sorry.\n'
```

Comment:
Is the trailing `\n` intentional? The input lacks one.

Reply:
Whoops.